src/main/java/ca/corefacility/bioinformatics/irida/repositories/relational/auditing/UserRevListener.java

```diff
@@ -1,93 +1,88 @@
+            User userByUsername = urepo.loadUserByUsername(username);
+
+            if (userByUsername != null) {
                 rev.setUserId(userByUsername.getId());
-            }
-            else{
+            } else {
                 throw new IllegalStateException("User could not be read by username so revision could not be created");
             }
-            
+
             //Add the client ID if the user is connected via OAuth2
             setClientId(rev);
-            
+
             logger.trace("Revision created by user " + userByUsername.getUsername());
-        }
-        catch(NullPointerException ex){
+        } catch (NullPointerException ex) {
             logger.error("No user is set in the session so it cannot be added to the revision.");
             throw new IllegalStateException("The database cannot be modified if a user is not logged in.");
         }
-        
-        
+
     }
 
     @Override
     public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
         UserRevListener.applicationContext = applicationContext;
     }
 
-	/**
-	 * Initialize the listener by getting dependencies
-	 */
-	public void initialize(){
+    /**
+     * Initialize the listener by getting dependencies
+     */
+    public void initialize() {
         urepo = applicationContext.getBean(UserRepository.class);
         clientRepo = applicationContext.getBean(IridaClientDetailsRepository.class);
     }
-    
-	/**
-	 * Add the OAuth2 client ID to the revision listener if the user is
-	 * connecting via OAuth2
-	 * 
-	 * @param entity
-	 *            The revision entity to modify if necessary
-	 */
-	private void setClientId(UserRevEntity entity) {
-		Authentication auth = SecurityContextHolder.getContext().getAuthentication();
-		// If the user is connecting via OAuth2 this object will be an
-		// OAuth2Authentication
-		if (auth instanceof OAuth2Authentication) {
-			try {
-				logger.trace("Found OAuth2Authentication in session.  Storing clientId in revision.");
-				OAuth2Authentication oAuth = (OAuth2Authentication) auth;
-				String clientId = oAuth.getOAuth2Request().getClientId();
-				IridaClientDetails clientDetails = clientRepo.loadClientDetailsByClientId(clientId);
-				entity.setClientId(clientDetails.getId());
-			} catch (NullPointerException ex) {
-				throw new IllegalStateException(
-						"The OAuth2 client details are not in the session so it cannot be added to the revision.");
-			}
-		}
-	}
-    
+
+    /**
+     * Add the OAuth2 client ID to the revision listener if the user is connecting via OAuth2
+     * 
+     * @param entity The revision entity to modify if necessary
+     */
+    private void setClientId(UserRevEntity entity) {
+        Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+        // If the user is connecting via OAuth2 this object will be a JwtAuthenticationToken
+        if (auth instanceof JwtAuthenticationToken) {
```

Comment:
If it is false it will not lead to an exception and it will not add the `client_id` to the Revision entity. A normal case of this would be when a user performs a db operation via the web interface. In that case use the `user_id` is set on the reivison entity.

Reply:
Okay, that makes sense to me.